flash_tutorials/electricity_forecasting/electricity_forecasting.py

```diff
@@ -0,0 +1,294 @@
+# breaks it down into a trend component and a seasonality component (in this case showing daily fluctuations in
+# electricity prices).
+#
+# ## Bonus: weekly trends
+#
+# The type of seasonality that the model learns to detect is dictated by the frequency of observations and the length of
+# the encoding / prediction window.
+# We might imagine that our pipeline could be changed to instead uncover weekly trends if we resample daily
+# observations from our data.
+#
+# We can use pandas to do this.
+# First, we load the data as before and create the index:
+
+# %%
+df_energy_daily = pd.read_csv("./data/energy_dataset.csv", parse_dates=["time"])
+
+df_energy_daily["time"] = pd.to_datetime(df_energy_daily["time"], utc=True, infer_datetime_format=True)
+df_energy_daily = df_energy_daily.set_index("time")
+
+# %% [markdown]
+# Next, we use the `resample` method to give us the mean value for each day (you could experiment with taking a max
+# here):
+
+# %%
+df_energy_daily = df_energy_daily.resample("D").mean()
+
+# %% [markdown]
+# Now let's create our `TabularForecastingData` as before, this time with an four week encoding window and a two week
+# prediction window.
+
+# %%
+df_energy_daily = df_energy_daily.filter(["price actual"])
+
+df_energy_daily["time_idx"] = (df_energy_daily.index.view(int) / (1e9 * 60 * 60 * 24)).astype(int)
+df_energy_daily["time_idx"] -= df_energy_daily["time_idx"].min()
+
+df_energy_daily["constant"] = 0
+
+max_prediction_length = 2 * 7
+max_encoder_length = 4 * 7
+
+training_cutoff = df_energy_daily["time_idx"].max() - max_prediction_length
+
+datamodule = TabularForecastingData.from_data_frame(
+    time_idx="time_idx",
+    target="price actual",
+    group_ids=["constant"],
+    max_encoder_length=max_encoder_length,
+    max_prediction_length=max_prediction_length,
+    time_varying_unknown_reals=["price actual"],
+    train_data_frame=df_energy_daily[lambda x: x.time_idx <= training_cutoff],
+    val_data_frame=df_energy_daily,
+    batch_size=256,
+)
+
+# %% [markdown]
+# Now it's time to create a new model and trainer.
+# We reduce the widths even further and run for many more epochs this time as we now have even fewer observations.
+# This time, instead of using the learning rate finder we just set the learning rate manually:
+
+# %%
+model = TabularForecaster(
+    datamodule.parameters,
+    backbone="n_beats",
+    backbone_kwargs={"widths": [8, 128], "backcast_loss_ratio": 0.1},
+    learning_rate=5e-4,
+)
+
+trainer = flash.Trainer(
+    max_epochs=3 * 24,
+    check_val_every_n_epoch=24,
+    gpus=1 if torch.cuda.is_available() else 0,
+    gradient_clip_val=0.01,
+)
+
+# %% [markdown]
+# Finally, we train the new model:
+
+# %%
+trainer.fit(model, datamodule=datamodule)
+
+# %% [markdown]
+# Now let's look at what it learned:
+
+# %%
+best_model_path = trainer.checkpoint_callback.best_model_path
```

Comment:
Might be worth to make a function of this, so you don't get duplicated code :) Same above.